WorkingWithText/Program.cs

```diff
@@ -7,29 +7,121 @@ namespace WorkingWithText
+            var result = true;
+
+            // checks to see if inputs are valid
+            foreach (var num in splitNum)
+                if (int.TryParse(num, out var parseInt))
+                    intList.Add(parseInt);
+                // if they're not then, that's as far as we go
+                else
+                    return false;
+
+            // now we cross check the count of distinct numbers with the count in the list
+            // if they dont match then there are duplicates, and we return false
+            result = intList.Distinct().Count() != intList.Count;
+
+            return result;
         }
 
+
         // 3- Write a method that accepts a string of a time 24-hour time format
         // (e.g. "09:00"). A valid time should be between 00:00 and 23:59. If the time is valid,
         // return a bool True; otherwise, return a bool False. If the user doesn't provide any values,
         // consider it as False. Make sure that its returns false if any letters are passed.
         public static bool IsValidTime(string hyphenNum)
         {
- 			return default;
+            var result = true;
+
+            // check the input to see if it contains stuff, if not return false
+            if (string.IsNullOrWhiteSpace(hyphenNum))
+            {
+                result = false;
+            }
+            // checks if first value is a 0, 1, or 2 because those are the only valid inital values for 24hr time
+            else if (hyphenNum[0] == '0' || hyphenNum[0] == '1' || hyphenNum[0] == '2')
+            {
+                // check to see if the given string is in the proper format 
+                if (hyphenNum.Contains(':'))
+                {
+                    var times = hyphenNum.Split(':');
+                    // now we make sure we don't crash by checking for valid number inputs
+                    var validHour = int.TryParse(times[0], out var hour);
+                    var validMin = int.TryParse(times[1], out var min);
+
+                    // if both are valid numbers then we'll check to see that they fall into the 24hr time format range
+                    if (validHour && validMin)
+                    {
+                        if (times.Length != 2)
+                            result = false;
+                        else if (hour < 0 || hour > 23)
+                            result = false;
+                        else if (min < 0 || min > 59)
+                            result = false;
+                    }
+                    // if inputs are not valid numbers
+                    else result = false;
+                }
+                // if there is no : as requested
+                else result = false;
+            }
+            // if the first value isn't a 1, 2 or 3
+            else
+            {
+                result = false;
+            }
+            // as long as result was never changed to false, we chillin
+            return result;
+
```

Comment:
This is a very good logical solution. Well thought out. The only thing I would warn you of is nesting so many if statements. Doing so can lead you down rabbit holes and breaks that you might not see. You could possibly avoid a lot of the nesting by using Boolean variables as flags to determine if it passes certain checks. Otherwise excellent work!